polaris-react/src/components/EmptyState/EmptyState.tsx

```diff
@@ -46,16 +46,30 @@ export function EmptyState({
   secondaryAction,
   footerContent,
 }: EmptyStateProps) {
-  const imageContainedClass = classNames(
+  const [imageLoaded, setImageLoaded] = useState<boolean>(false);
+
+  const onImageLoad = useCallback(() => {
+    setImageLoaded(true);
+  }, []);
+
+  useEffect(() => {
+    const img: HTMLImageElement = new window.Image(0, 0);
+    img.src = largeImage || image;
+    img.onload = () => onImageLoad();
+  }, [onImageLoad, largeImage, image]);
```

Comment:
Instead of this useEffect and creating a new image (not sure if this kicks off a second network request), can we just use the `onLoad` prop on `Image`?